draft-ietf-jsonpath-base.md

```diff
@@ -1502,10 +1502,21 @@ The well-typedness of function expressions can now be defined in terms of this t
 
 For a function expression to be well typed:
 
-1. its declared type must be well typed in the context in which it occurs, and
-2. its arguments must be well typed for the declared type of the corresponding parameters.
+1. its function name must be registered (see {{iana-fnex}}),
+2. its declared type must be well typed in the context in which it occurs, and
+3. its arguments must be well typed for the declared type of the corresponding parameters.
 
-(1) As per the grammar, a function expression can occur in three different
+(1) The registered set of function extensions is the source of knowledge regarding the return and
```

Comment:
```suggestion
(1) The registered set of function extensions is the source of knowledge regarding the declared return and
```

I like saying that these are declared (as opposed to actual, as in a dynamic type system).